### PR TITLE
First version of compress and estimate gain

### PR DESCRIPTION
**Pull request by delarosatrevin**
Added two protocols related to compression (both using relion_convert_to_tif):
* relion - compress movie
* relion - compress estimate gain

I haven't created any test since it will required original movies and can be quite heavy, but I'm open to suggestions.

### CODE REVIEW

**Comment by azazellochg** [comment]
I find it a bit strange: compress protocol without estimation option and compress estimate protocol without compressing. I think it would make sense to do:
1) estimate protocol
2) compress protocol with optional gain estimation

**Comment by delarosatrevin** [comment]
@azazellochg 
1) My first approach was to name just 'estimate gain' but it seems that there is a another Relion program that is used for gain estimation. That's why I just kept the 'compress' prefix but I know it is confusing. 
2) The input gain is already optional in the compress protocol.

**Comment by azazellochg** [comment]
@delarosatrevin true, there is a separate estimation program. Then maybe we need only 1 protocol compress which has optional: input gain file or option to estimate gain and then compress

**Comment by azazellochg** [comment]
Also, since gain is set during import movies protocol we can remove option 1, just use associated gain by default or estimate it in the program

**Comment by delarosatrevin** [comment]
Hum...not sure about the relation of this gain and the one used in the import...if they are equivalent or if this one is more Relion-specific (even the .bin format). I also tried to put both estimation and compression in the same protocol but finally keep it separated. There are cases where you want to compress, without gain, and not need to estimate it:
https://www3.mrc-lmb.cam.ac.uk/relion/index.php/Image_compression

File: relion/protocols/protocol_compress_estimate_gain.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+from pwem.protocols import ProtProcessMovies
+from pyworkflow.protocol import STEPS_SERIAL
+
+import relion
+import relion.convert as convert
+
+
+class ProtRelionCompressEstimateGain(ProtProcessMovies):
+    """
+    Using *relion_convert_to_tiff to compress a set of movies
```
Comment (azazellochg) [inline relion/protocols/protocol_compress_estimate_gain.py:38]: missing second asterisk

File: relion/protocols/protocol_compress_estimate_gain.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    or to estimate the gain that can be used for better compression.
+    """
+
+    _label = 'compress estimate gain'
```
Comment (azazellochg) [inline relion/protocols/protocol_compress_estimate_gain.py:42]: maybe estimate gain for compression?

File: relion/protocols/protocol_compress_estimate_gain.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+    def __init__(self, **kwargs):
+        ProtProcessMovies.__init__(self, **kwargs)
+        self.stepsExecutionMode = STEPS_SERIAL
+
+    def _getConvertExtension(self, filename):
+        """ Check whether it is needed to convert to .mrc or not """
+        ext = pwutils.getExt(filename).lower()
+        return None if ext in ['.mrc', '.mrcs', '.tiff', '.tif'] else 'mrc'
```
Comment (azazellochg) [inline relion/protocols/protocol_compress_estimate_gain.py:51]: add a check if movie is in tif there is no point using this protocol

File: relion/protocols/protocol_compress_estimate_gain.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    # --------------------------- INFO functions ------------------------------
+    def _summary(self):
+        summary = []
+        return summary
+
+    def _citations(self):
+        return ['Zivanov2019']
+
+    def _validate(self):
+        # Check base validation before the specific ones for Motioncor
```
Comment (azazellochg) [inline relion/protocols/protocol_compress_estimate_gain.py:117]: ???

File: relion/protocols/protocol_compress_movies.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+from pwem.objects import Image, MovieAlignment
+from pyworkflow.protocol import STEPS_PARALLEL
+
+import relion
+
+
+class ProtRelionCompressMovies(ProtAlignMovies):
+    """
+    Using *relion_convert_to_tiff to compress a set of movies
+    or to estimate the gain that can be used for better compression.
```
Comment (azazellochg) [inline relion/protocols/protocol_compress_movies.py:41]: better: Using *relion_convert_to_tiff* to compress a set of movies

File: relion/protocols/protocol_compress_movies.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+    def __init__(self, **kwargs):
+
+        ProtAlignMovies.__init__(self, **kwargs)
+        self.stepsExecutionMode = STEPS_PARALLEL
+
+    def _getConvertExtension(self, filename):
+        """ Check whether it is needed to convert to .mrc or not """
+        ext = pwutils.getExt(filename).lower()
+        return None if ext in ['.mrc', '.mrcs', '.tiff', '.tif'] else 'mrc'
```
Comment (azazellochg) [inline relion/protocols/protocol_compress_movies.py:57]: add tif check as above

File: relion/protocols/protocol_compress_movies.py [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+    def _summary(self):
+        summary = []
+        return summary
+
+    def _citations(self):
+        return ['Zivanov2019']
+
+    def _validate(self):
+        # Check base validation before the specific ones for Motioncor
+        errors = ProtAlignMovies._validate(self)
```
Comment (azazellochg) [inline relion/protocols/protocol_compress_movies.py:122]: why alignmovies here?

**Comment by azazellochg** [comment]
@delarosatrevin you are right, after reading the wiki I understand your idea behind two protocols. Just few comments above.

File: relion/protocols/protocol_compress_movies.py (hunk 7 above)
Comment (delarosatrevin) [reply relion/protocols/protocol_compress_movies.py:122]: Despite it is not purely an "AlignMovies" protocol, it has some similarities. I mainly used it to re-use the streaming infrastructure.

File: relion/protocols/protocol_compress_movies.py (hunk 6 above)
Comment (delarosatrevin) [reply relion/protocols/protocol_compress_movies.py:57]: Not completely sure, but I guess tiff images can also be compressed and written as another tiff file.

File: relion/protocols/protocol_compress_estimate_gain.py (hunk 4 above)
Comment (delarosatrevin) [reply relion/protocols/protocol_compress_estimate_gain.py:117]: Left overs, now removed

**Comment by delarosatrevin** [comment]
@azazellochg I have addressed most of your comments now.

**Pull request merged**